common/crypto/verify_ias_report/verify-report.cpp

```diff
@@ -27,6 +27,16 @@
 //########### INTERNAL FUNCTIONS #########################################
 //########################################################################
 
+#define COND2ERR(b) \
+    do \
```

Comment:
Regarding the do/while, no but, in general, it might be useful in other situations.
Regarding the if block, no, but it's good practice.

Reply:
What other circumstances would it be where the do/while block is necessary?  putting a (simple) block in macros around everything can help if, say the invoker defines variables or alike so it would be scoped. But the if itself already does scope via condition paranthesis and body. So don't see even  a use-case where the empty block helps.  But the empty block i have less of an option, that clutters/confuses less than the do/while ....

Reply:
see nice explanation here https://www.quora.com/What-is-the-purpose-of-using-do-while-0-in-macros

Reply:
thanks for the pointer. While in this case you do not really need to do/while, putting just curly brackets around as i first suggested would give you the if/else problem and only no brackets would work (given that the `if` has the same effect, in fact an `if (true) { ... };` could be used as a similar generic pattern?).   So yes, do leave it

Reply:
by the way... do {} while() is a pretty standard way to keep macros clean. if you expect it to be a statement then its a very good idea otherwise you can end up with ambiguous if/then if the macro is embedded in the statement.